libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/websocket/WebsocketRouteAdaptor.kt

```diff
@@ -96,7 +109,8 @@ internal class WebsocketRouteAdaptor(
     // The handler is called when a WebSocket client closes the connection.
     override fun handleClose(ctx: WsCloseContext) {
         try {
-            requireNotNull(channel).onClose?.let { it(ctx.status(), ctx.reason()) }
+            requireNotNull(channels[ctx.sessionId]).onClose?.invoke(ctx.status(), ctx.reason())
+            channels.remove(ctx.sessionId)
```

Comment:
Should we be a bit more lenient here to ensure we handle call to `handleClose` twice, i.e.:
```suggestion
            channels.remove(ctx.sessionId)?.onClose?.invoke(ctx.status(), ctx.reason())
```